include/vctr/Expressions/Core/Clamp.h

```diff
@@ -0,0 +1,269 @@
+    // Platform Vector Operation Implementation
+    VCTR_FORCEDINLINE const value_type* evalNextVectorOpInExpressionChain (value_type* dst) const
+    requires is::suitableForAccelerateRealFloatVectorOp<SrcType, value_type, detail::dontPreferIfIppAndAccelerateAreAvailable>
+    {
+        if constexpr (clampLow && ! clampHigh)
+            Expression::Accelerate::clampLow (src.evalNextVectorOpInExpressionChain (dst), lowerBound, dst, size());
+
+        if constexpr (clampHigh)
+        {
+            auto l = clampLow ? lowerBound : std::numeric_limits<value_type>::lowest();
+            Expression::Accelerate::clamp (src.evalNextVectorOpInExpressionChain (dst), l, upperBound, dst, size());
+        }
+
+        return dst;
+    }
+
+    VCTR_FORCEDINLINE const value_type* evalNextVectorOpInExpressionChain (value_type* dst) const
+    requires is::suitableForIppRealFloatVectorOp<SrcType, value_type, detail::preferIfIppAndAccelerateAreAvailable>
+    {
+        const auto* s = src.evalNextVectorOpInExpressionChain (dst);
+
+        if constexpr (clampLow)
+        {
+            Expression::IPP::clampLow (s, lowerBound, dst, sizeToInt (size()));
+
+            if constexpr (clampHigh)
+                s = dst;
+        }
+
+        if constexpr (clampHigh)
+            Expression::IPP::clampHigh (s, upperBound, dst, sizeToInt (size()));
+
+        return dst;
+    }
+
+private:
+    SrcType src;
+};
+
+} // namespace vctr::Expressions
+
+namespace vctr
+{
+
+/** Ensures that the elements are not lower than lowerBound.
+
+    This version takes a runtime computed value. If a compile time fix constant lower bound suits your needs,
+    consider clampLowByConstant.
+
+    @ingroup Expressions
+ */
+template <class T>
+constexpr auto clampLow (T lowerBound)
+{
+    return makeExpressionChainBuilderWithRuntimeArgs<Expressions::ClampLow> (lowerBound, std::numeric_limits<T>::max());
+}
+
+/** Ensures that the elements are not greater than upperBound.
+
+    This version takes a runtime computed value. If a compile time fix constant lower bound suits your needs,
+    consider clampHighByConstant.
+
+    @ingroup Expressions
+ */
+template <class T>
+constexpr auto clampHigh (T upperBound)
+{
+    return makeExpressionChainBuilderWithRuntimeArgs<Expressions::ClampHigh> (std::numeric_limits<T>::max(), upperBound);
+}
+
+/** Ensures that the elements are not lower than lowerBound and not higher than upperBound.
+
+    This version takes runtime computed values. If compile time fix constants suit your needs,
+    consider clampByConstant.
+
+    @ingroup Expressions
+ */
+template <class T>
+constexpr auto clamp (T lowerBound, T upperBound)
+{
+    return makeExpressionChainBuilderWithRuntimeArgs<Expressions::ClampLowHigh> (lowerBound, upperBound);
+}
+
+
+
+/** Ensures that the elements are not lower than lowerBound.
+
+    Note: At this time, most compilers wil refuse to take floating point non-type template values.
```

Comment:
*will refuse. Also two instances below 🙂

Reply:
Ah yes, will fix that